connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/bop/BOPManager.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.cloud.notifications.connector.email.processors.bop;
+
+import com.redhat.cloud.notifications.connector.email.config.EmailConnectorConfig;
+import com.redhat.cloud.notifications.connector.email.model.bop.Email;
+import com.redhat.cloud.notifications.connector.email.model.bop.SendEmailsRequest;
+import dev.failsafe.Failsafe;
+import dev.failsafe.RetryPolicy;
+import dev.failsafe.function.CheckedRunnable;
+import io.quarkus.logging.Log;
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Set;
+
+@ApplicationScoped
+public class BOPManager {
+
+    public static final String NOTIFICATIONS_BOP_RETRY_MAX_ATTEMPTS = "notifications.bop.retry.max-attempts";
+    public static final String NOTIFICATIONS_BOP_RETRY_INITIAL_BACKOFF = "notifications.bop.retry.initial-backoff";
+    public static final String NOTIFICATIONS_BOP_RETRY_MAX_BACKOFF = "notifications.bop.retry.max-backoff";
+
+    @ConfigProperty(name = NOTIFICATIONS_BOP_RETRY_MAX_ATTEMPTS, defaultValue = "3")
+    int maxRetryAttempts;
+
+    @ConfigProperty(name = NOTIFICATIONS_BOP_RETRY_INITIAL_BACKOFF, defaultValue = "0.1S")
+    Duration initialRetryBackoff;
+
+    @ConfigProperty(name = NOTIFICATIONS_BOP_RETRY_MAX_BACKOFF, defaultValue = "1S")
+    Duration maxRetryBackoff;
+
+    private RetryPolicy<Object> retryPolicy;
+
+    @Inject
+    @RestClient
+    BOPService bopService;
+
+    @Inject
+    EmailConnectorConfig emailConnectorConfig;
+
+    @PostConstruct
+    public void postConstruct() {
+        retryPolicy = RetryPolicy.builder()
+            .handle(IOException.class)
+            .withBackoff(initialRetryBackoff, maxRetryBackoff)
+            .withMaxAttempts(maxRetryAttempts)
+            .onRetriesExceeded(event ->
+                // All retry attempts failed, let's log a warning about the failure.
+                Log.warn("Sending email to BOP service failed", event.getException())
+            )
+            .build();
```

Comment:
Does going over the retries limit also throw an exception? I am wondering what happens if we fail to contact BOP: would that throw an exception and get caught by the exception processor, which would mark this notification as "failed" in the notification history?

Reply:
That's the idea yes, because I did it few weeks ago, I'm no more 100% (only 90% :) ). Let me rebase and retest it locally to confirm that.

Reply:
I confirm, it works as you described.
I also added `jakarta.ws.rs.ProcessorException`, because it can be use to propagate IO exception as mentionned in its [javadoc](https://jakarta.ee/specifications/restful-ws/3.0/apidocs/jakarta/ws/rs/processingexception)

Reply:
Thank you for confirming it!